File: src/test_time_measurment.cpp
#include <ros/ros.h>

int main(int argc, char** argv)
{
	ros::init(argc, argv, "mesureing_computation_time");
	ros::NodeHandle nh;

	ros::Time t_start = ros::Time::now();

	/*Start*/
	for(int i=0; i<10000; ++i){
		std::cout << "i = " << i << std::endl;
		std::cout << "time[s] = " << (ros::Time::now() - t_start).toSec() << std::endl;
	}
	/*End*/

	std::cout << "Computation is done" << std::endl;
	std::cout << "Computation time[s] = " << (ros::Time::now() - t_start).toSec() << std::endl;

	return 0;
}

File: pysrc/test_time_measurment.py
#!/usr/bin/env python

import rospy

def main():
    rospy.init_node('mesureing_computation_time', anonymous=True)

    t_start = rospy.get_time()

    ## Start
    for i in range(10000):
        print("i = ", i)
        print("time[s] = ", rospy.get_time() - t_start)
    ## End

    print("Computation is done")
    print("Computation time[s] = ", rospy.get_time() - t_start)

if __name__ == '__main__':
    main()
